docs/deployment-gcp.md

```diff
@@ -0,0 +1,134 @@
+gcloud artifacts repositories create {repository-name} \
+    --repository-format docker \
+    --location us-west1 \
+    --description "Docker repository for storing images of Canopy."
+```
+
+## Step 3: Pull and Tag Canopy Docker Image
+We'll start by fetching the official `Canopy` image from GitHub Packages, and then we'll apply a tag to prepare the image
+for pushing it to Google Cloud Platform (`GCP`). You can access all the available images [here](https://github.com/pinecone-io/canopy/pkgs/container/canopy).
+
+```bash
+# Pull the Canopy Docker image
+docker pull ghcr.io/pinecone-io/canopy:{canopy-version}
+
+# Tag the image for GCP repository
+docker tag ghcr.io/pinecone-io/canopy:{canopy-version} us-west1-docker.pkg.dev/{project-name}/{repository-name}/canopy:{canopy-version}
+```
+
+## Step 4: Configure Docker for GCP
+```bash
+# Configure Docker to use GCP credentials
+gcloud auth configure-docker us-west1-docker.pkg.dev
+```
+
+## Step 5: Push Canopy Docker Image to GCP
+```bash
+# Push the Canopy Docker image to GCP repository
+docker push us-west1-docker.pkg.dev/{project-name}/{repository-name}/canopy:{canopy-version}
+```
+
+## Step 6: Deploy Canopy on Google Cloud Run
+Before running the following command make sure to create a `.env` file and include the environment variables mentioned
+in [README.md](https://github.com/pinecone-io/canopy/blob/main/README.md). 
+```text
+OPENAI_API_KEY={open-api-key}
+PINECONE_API_KEY={pinecone-api-key}
+PINECONE_ENVIRONMENT={pinecone-environment}
+
+# Other necessary environment variables if needed
+```
+
+In the same directory, run the following command:
+
+```bash
+# Deploy Canopy on Google Cloud Run
+gcloud run deploy canopy \
+  --image us-west1-docker.pkg.dev/{project-name}/{repository-name}/canopy:{canopy-version} \
+  --platform managed \
+  --region us-west1 \
+  --min-instances 1 \
+  --port 8000 \
+  --allow-unauthenticated \
+  --set-env-vars $(grep -v '^#' .env | tr '\n' ',' | sed 's/,$//')
+```
+
+Congratulations! You have successfully deployed `Canopy` on `GCP`. 
+
+You should now see an output similar to this:
+
+    Deploying container to Cloud Run service [canopy] in project [project-name] region [us-west1]
+    ✓ Deploying new service... Done.
+      ✓ Creating Revision...
+      ✓ Routing traffic...
+      ✓ Setting IAM Policy...
+    Done.
+    Service [canopy] revision [canopy-00001-6cf] has been deployed and is serving 100 percent of traffic.
+    Service URL: https://canopy-bxkpka-uw.a.run.app
+
+## Step 7 - Testing Access to the Service
+From your terminal run the following command using the service url you have received from `GCP`:
+
+```bash
+curl {service-url}/v1/health
+```
+
+If you see the following output your deployment is completed successfully!
+
+```json
+{"pinecone_status":"OK","llm_status":"OK"}
+```
+
+## Step 8 - Chatting with the Service (Optional)
+
+In order to chat with the service, clone the `Canopy` project and install the dependencies using `Poetry`.
```

Comment:
This section should actually say - you can use any Open-AI compatible chat UI to interact with the newly deployed server. Examples: (links to some OSS chat UIs)

If you want to use OpenAI's python client see https://github.com/pinecone-io/canopy?tab=readme-ov-file#migrating-an-existing-openai-application-to-canopy

If you want to use Canopy's built-in Chat CLI, you can run
```bash
docker run ghcr.io/pinecone-io/canopy:{canopy-version} canopy chat
```
(The user just pulled our docker image - why would he need to `pip install`...?)

Reply:
Right no need to pip install :) Changing to reflect the chat UI

Reply:
Done